homeassistant/components/frontier_silicon/media_player.py

```diff
@@ -205,6 +221,15 @@ async def async_update(self):
 
             self._attr_volume_level = None
 
+    @property
+    def supported_features(self) -> int:
+        """Return media player features that are supported."""
+        result = self._always_supported_features
+        if self._supports_sound_mode:
+            result |= MediaPlayerEntityFeature.SELECT_SOUND_MODE
+
+        return result
+
```

Comment:
I think you should move all this logic inside `update` method, and set/update self._attr_supported_features there instead.

I think there are also restrictions that say supported_features should not be dynamically adjusted and should instead be defined at startup, but I'm not 100% sure about that so better wait for confirmation.

Reply:
> I think there are also restrictions that say supported_features should not be dynamically adjusted and should instead be defined at startup, but I'm not 100% sure about that so better wait for confirmation.

Yeah preferably, however for media players we are a little less strict on that one (the frontend can handle it and sometimes capabilities differ based on things like select source and such).

Reply:
While I agree that it would be nicer, moving it to the setup might prove problematic as the frontier_silicon devices do not allow to query the API when the device is turned off.

So if it is allowed to change the `supported_features` on update I would like to keep it like this. I will also need to do this for hiding the play/pause feature when a radio source is selected.